tools/bumper/parser.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"reflect"
+
+	"github.com/pkg/errors"
+
+	"gopkg.in/yaml.v2"
+)
+
+type metadata struct {
+	Tag		string `yaml:"tag"`
+}
+
+type component struct {
+	Url         string `yaml:"url"`
+	Commit      string `yaml:"commit"`
+	Metadata 	metadata
+}
+
+type ComponentsList struct {
+	Multus component `yaml:"multus"`
+	LinuxBridge component `yaml:"linux-bridge"`
+	BridgeMarker component `yaml:"bridge-marker"`
+	Kubemacpool component `yaml:"kubemacpool"`
+	Nmstate component `yaml:"nmstate"`
+	OvsCni component `yaml:"ovs-cni"`
+	MacvtapCni component `yaml:"macvtap-cni"`
+}
+
+type ComponentsConfig struct {
+	Components ComponentsList
+}
+
+func parseComponentsYaml(componentsConfigPath string) (map[string]component, error) {
+	config := ComponentsConfig{}
+
+	Logger.Printf("Opening components config %s", componentsConfigPath)
+	componentsData, err := ioutil.ReadFile(componentsConfigPath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to open file %s", componentsConfigPath)
+	}
+
+	Logger.Printf("Unmarshaling components config %s", componentsConfigPath)
+	err = yaml.Unmarshal(componentsData, &config)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to Unmarshal %s", componentsConfigPath)
+	}
+
+	if config == (ComponentsConfig{}) {
```

Comment:
This covers `len(config.components) == 0` ?

Reply:
change to be explicit. Done